pkg/reconciler/v1alpha1/revision/resources/deploy.go

```diff
@@ -119,6 +119,7 @@ func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, observab
 	userContainer.VolumeMounts = append(userContainer.VolumeMounts, varLogVolumeMount)
 	userContainer.Lifecycle = userLifecycle
 	userContainer.Env = append(userContainer.Env, userEnv)
+	userContainer.Env = append(userContainer.Env, GetKnativeEnvVar(rev)...)
```

Comment:
Do we want to overwrite the user's env vars, or have the user be able to override the Knative environment variables?

(I don't know; this code appears to override the user's settings, but I could see an argument for the other way, too.)

Reply:
Its usually recommended that the user does not using environment variables with the `K_` or `KNATIVE` prefix or risk these variables being overwritten

Reply:
For the `service_name` if its empty, its currently getting set to empty string if its empty which I think would be what we would want to have anyways